pkg/controllers/streaming/processor_controller.go

```diff
@@ -422,6 +426,53 @@ func (r *ProcessorReconciler) constructDeploymentForProcessor(processor *streami
 		return nil, err
 	}
 
+	volumes := []corev1.Volume{}
+	volumeMounts := []corev1.VolumeMount{}
+	for _, stream := range streams {
+		if stream.Status.Binding.MetadataRef.Name != "" {
+			metadataVolumeName := fmt.Sprintf("processor-stream-%s-metadata", stream.Name)
+			volumes = append(volumes,
+				corev1.Volume{
+					Name: metadataVolumeName,
+					VolumeSource: corev1.VolumeSource{
+						ConfigMap: &corev1.ConfigMapVolumeSource{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: stream.Status.Binding.MetadataRef.Name,
+							},
+						},
+					},
+				},
+			)
+			volumeMounts = append(volumeMounts,
+				corev1.VolumeMount{
+					Name:      metadataVolumeName,
```

Comment:
Same remark as above, with even the additional comment/discussion below:

shouldn't the "role" played by the stream appear here somehow, ie "first input" (input0), "second input" (input1), etc?
It seems otherwise we need an extra info anyway, I guess conveyed by an ENV VAR.

Reply:
See above comment. The name of the volume is not exposed inside the container.

> shouldn't the "role" played by the stream appear here somehow, ie "first input" (input0), "second input" (input1), etc?

I didn't want to introduce an encoding scheme into the name of the binding.

> It seems otherwise we need an extra info anyway, I guess conveyed by an ENV VAR.

Bingo. That's why I didn't mark the INPUT_NAMES and OUTPUT_NAMES envvar for removal.

Reply:
> The name of the volume is not exposed inside the container

I was thinking more of part of the mount path.

> Bingo. That's why I didn't mark the INPUT_NAMES and OUTPUT_NAMES envvar for removal.

That's a misunderstanding of what INPUT/OUTPUT_NAMES are. They're aliases, logical names that appear in the target function code (eg keys in a map, names of fields in a struct, etc) whose _default_ value come from stream names, but they're NOT stream names.

So it seems we'll always need extra info that the binding spec doesn't address. The question is where do we (and are _we_ the ones to decide, or is it actually mandated in the spec, really? this is why I wanted to discuss this subject further in the RFC, but discussing it over actual running code is fine too) draw the line.

Reply:
OK, the binding name should be using the alias rather than the stream's name. I'll fix that.

> So it seems we'll always need extra info that the binding spec doesn't address.

That's by design. The binding is not responsible for carying metadata about how the binding should be consumed, it only contains metadata about the provisioned resource that can be accessed. Metadata about how the resource should be consumed must be passed outside of the binding. The spec says:

> The binding name is the only aspect of the binding that is under the control of the consumer. All other elements of the binding are defined by the binding producer. Additional metadata may be injected into the workload about the binding out of band, but is not part of this RFC.